File: source/downloads/code/LeetCode/Python/Gas-Station.py
class Solution(object):
    def canCompleteCircuit(self, gas, cost):
        """
        :type gas: List[int]
        :type cost: List[int]
        :rtype: int
        """
        bal = 0
        start = 0
        for i in range(len(gas)):
            bal += gas[i] - cost[i]
            if bal < 0:
                start = i+1
                bal = 0
        if sum(gas) - sum(cost) >=0:
            return start
        else:
            return -1

File: downloads/code/LeetCode/Java/Gas-Station.java
public class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int start = 0;
        int balance = 0;
        int totalGas = 0;
        int totalCost = 0;
        for(int i=0;i<gas.length;i++){
            totalGal += gas[i];
            totalCost += cost[i];
            balance += gas[i]-cost[i];
            if(balance < 0){
                start = i+1;
                balance = 0;
            }
        }
        if(totalGas>=totalCost) return start;
        else return -1;    
    }
}
